### DISCUSSION TITLE
Smarter syntax highlighting

### DISCUSSION THREAD

**Discussion by ReZLone** [lzybkr/PSReadLine#4110] [Ideas]
I saw that clink for cmd is able to give a smarter input line syntax highlighting by using different colors based on the command typed (ex. if the command can't be runned or in other words doesn't exist it would be written in red).
I think something like this in powershell would be a great addition. There should be different colors for aliases,cmdlets and executables and a real time validation/linting that if fails will color all in red.

**Comment by digitalguy99** [comment]
A module that does this already exists. It's not perfect but definitely worth taking a look at. Check it out [here](https://github.com/digitalguy99/pwsh-syntax-highlighting).

Tldr, simply run the following:

1. (to install module)
	```pwsh
	Install-Module syntax-highlighting
	```
2. (so that you won't have import the module every time you run PowerShell)
	```pwsh
	echo "Import-Module syntax-highlighting" >> $profile
	```
